### PR TITLE
Add MVP suggestion for `unsafe_op_in_unsafe_fn`

### PR DESCRIPTION
**Pull request by Nemo157**
Rebase of https://github.com/rust-lang/rust/pull/99827

cc tracking issue https://github.com/rust-lang/rust/issues/71668

No real changes since the original PR, just migrated the new suggestion to use fluent messages and added a couple more testcases, AFAICT from the discussion there were no outstanding changes requested.

### CODE REVIEW

File: tests/ui/unsafe/rfc-2585-unsafe_op_in_unsafe_fn.mir.stderr [hunk 1]
```diff
@@ -10,6 +10,14 @@ note: the lint level is defined here
    |
 LL | #![deny(unsafe_op_in_unsafe_fn)]
    |         ^^^^^^^^^^^^^^^^^^^^^^
+help: consider wrapping the function in an unsafe block
```
[the remark this answers is not in the document]
Comment (Mark-Simulacrum) [reply tests/ui/unsafe/rfc-2585-unsafe_op_in_unsafe_fn.mir.stderr:13]: The existing suggestion I gave: not providing a in-code suggestion and instead provide just the help text, e.g., "help: an unsafe function restricts callers, but it's body is safe by default".

File: compiler/rustc_mir_transform/src/errors.rs [hunk 2]
```diff
@@ -138,13 +140,20 @@ impl<'a> DecorateLint<'a, ()> for UnsafeOpInUnsafeFn {
...
-            .eagerly_translate_to_string(self.details.label(), [].into_iter());
+        let handler = diag.handler().expect("lint should not yet be emitted");
+        let desc = handler.eagerly_translate_to_string(self.details.label(), [].into_iter());
         diag.set_arg("details", desc);
         diag.span_label(self.details.span, self.details.label());
         diag.note(self.details.note());
+
+        if let Some((start, end)) = self.suggest_unsafe_block {
+            diag.multipart_suggestion_verbose(
+                crate::fluent_generated::_subdiag::suggestion,
```
Comment (Nemo157) [inline compiler/rustc_mir_transform/src/errors.rs:151]: Ah, also I assume using `_subdiag` here is wrong, but stuff around the fluent messages seems to have changed compared to the rustc-dev-guide, so I wasn't able to figure out how to access the subdiagnostics correctly.

Comment (Nemo157) [reply compiler/rustc_mir_transform/src/errors.rs]: Found the solution by checking the macro impl, it adds the crate prefix onto the labels.

File: compiler/rustc_mir_transform/src/errors.rs [hunk 3]
```diff
@@ -130,6 +131,7 @@ impl RequiresUnsafeDetail {
 
 pub(crate) struct UnsafeOpInUnsafeFn {
     pub details: RequiresUnsafeDetail,
+    pub suggest_unsafe_block: Option<(Span, Span, Span)>,
```
Comment (eholk) [inline compiler/rustc_mir_transform/src/errors.rs:134]: Maybe add a doc comment describing what each of the three spans means?

**Review by eholk — APPROVED** [review]
Looks good. I made a few minor suggestions, which you can take or leave at your discretion. Feel free to r=me once you're ready.

**Pull request merged**